### DISCUSSION TITLE
How can I add my own name env. to base_case and start it from CLI

### DISCUSSION THREAD

**Discussion by SergeyGuschin** [seleniumbase/SeleniumBase#1806] [General]
Hello Michael. I have a question for you. How can I add my own env name? So I can type the command in CLI --env=myownname and get it to work? I try to modify base_case file by it still gives me an error. 

**base_case.py**

![image](https://user-images.githubusercontent.com/96178944/225724617-b2b4ab85-7054-435b-a2bb-3b6d76205857.png)

**The error**
![image](https://user-images.githubusercontent.com/96178944/225726402-bed726fe-b94d-465c-8bce-825ae9b02975.png)

**Comment by mdmintz** [comment]
Hi @SergeyGuschin, Based on your screenshots, it looks like you're mixing up valid pytest markers with valid envs. ``ValidEnvs`` is defined here: https://github.com/seleniumbase/SeleniumBase/blob/master/seleniumbase/fixtures/constants.py (That can only be changed from a cloned copy of SeleniumBase). Envs are picked from a list to prevent issues that could result when misspelling env names. You could use a different variable that isn't checked from a list, such as ``--data=DATA`` (``self.data``) or ``--var1=VAR1`` (``self.var1``), and others:

```bash
--account=STR  (Set account. Access with "self.account" in tests.)
--data=STRING  (Extra test data. Access with "self.data" in tests.)
--var1=STRING  (Extra test data. Access with "self.var1" in tests.)
--var2=STRING  (Extra test data. Access with "self.var2" in tests.)
--var3=STRING  (Extra test data. Access with "self.var3" in tests.)
--variables=DICT  (Extra test data. Access with "self.variables".)
```

(That can be found in https://github.com/seleniumbase/SeleniumBase/blob/master/seleniumbase/plugins/pytest_plugin.py)

The ``--env=ENV`` one is special, and is checked for a match in the ``ValidEnvs`` list from ``fixtures/constants.py``.

**Reply by SergeyGuschin** [reply]
Thank a lot Michael, for your quick answer and tip. Now I've got it working as I want. I've changed the variables' names as I prefer.
